motionplan/ik/metrics.go

```diff
@@ -10,6 +10,26 @@ import (
 
 const orientationDistanceScaling = 10.
 
+// Segment contains all the information a constraint needs to determine validity for a movement.
+// It contains the starting inputs, the ending inputs, corresponding poses, and the frame it refers to.
+// Pose fields may be empty, and may be filled in by a constraint that needs them.
+type Segment struct {
```

Comment:
This file doesn't make a ton of sense in this package.  I almost think it belongs in referenceframe

Reply:
It's not used at all by anything in `referenceframe` and is only used in `motionplan` and its two subpackages; I'm inclined to leave it here.

At a point in the future we may want to rework how we treat Inputs, States, Segments, Nodes, etc and what gets passed where.

Reply:
I think just reading `ik.Segment` a number of times jumps out to me as a little fishy. What relates this structure to IK?  I'm ok to leave it here for now but starting to think about where the above things should ultimately live would be a good thing

Reply:
Part of that is, planning and solving in terms of segments lends itself rather nicely to PTGs and is something that's on my mind for implementing in the future.

But at the end of the day, it might well exist more nicely in `referenceframe` as well. I'm open to that. I think I would be more inclined if the word "referenceframe" wasn't already so darn long :)